caas/kubernetes/provider/k8s.go

```diff
@@ -246,22 +243,29 @@ func (k *kubernetesClient) ensureNamespaceAnnotationForControllerUUID(controller
 	if err != nil && !errors.IsNotFound(err) {
 		return errors.Trace(err)
 	}
+	// If the namespace doesn't exist then there is nothing we can do.
+	if errors.IsNotFound(err) {
+		return nil
+	}
 	annotationControllerUUIDKey := utils.AnnotationControllerUUIDKey(isLegacy)
+
+	// If juju doesn't own the namespace then we should leave it alone.
+	if ns != nil && !isK8sObjectOwnedByJuju(ns.ObjectMeta) {
+		return nil
+	}
+
 	if !isLegacy {
 		// Ignore the controller uuid since it is handled below for model migrations.
 		expected := k.annotations.Copy()
 		expected.Remove(annotationControllerUUIDKey)
 		if ns != nil && !k8sannotations.New(ns.Annotations).HasAll(expected) {
 			// This should never happen unless we changed annotations for a new juju version.
 			// But in this case, we should have already managed to fix it in upgrade steps.
-			return errors.NewNotValid(nil,
-				fmt.Sprintf("annotations %v for namespace %q must include %v", ns.Annotations, k.namespace, k.annotations),
-			)
+			return fmt.Errorf(
+				"model namespace %q does not included the expected annotations %v for this controller.",
```

Comment:
typo
can we keep the original error message, it has more detail, eg the actual annotations